File: listr-svc/ListR/ListR.Common/Models/Auth/RegisterModel.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ListR.Common.Models.Auth
{
    public class RegisterModel
    {
        public string? Username { get; set; }

        public string? Email { get; set; }

        public string? FirstName { get; set; }

        public string? LastName { get; set; }

        public string? Password { get; set; }
    }
}

File: listr_app/ListR/lib/common/models/Requests/register_model.dart
class RegisterModel {
  String? username;
  String? email;
  String? firstName;
  String? lastName;
  String? password;

  RegisterModel(
      {this.username,
      this.email,
      this.firstName,
      this.lastName,
      this.password});

  RegisterModel.fromJson(Map<String, dynamic> json) {
    username = json['username'];
    email = json['email'];
    firstName = json['firstName'];
    lastName = json['lastName'];
    password = json['password'];
  }

  Map<String, dynamic> toMap() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['username'] = username;
    data['email'] = email;
    data['firstName'] = firstName;
    data['lastName'] = lastName;
    data['password'] = password;
    return data;
  }
}
